src/engine/strat_engine/pool.rs

```diff
@@ -151,9 +151,16 @@ impl Pool for StratPool {
         // TODO: Roll back on filesystem initialization failure.
         let dm = DM::new()?;
         let mut result = Vec::new();
-        for name in names {
-            let fs_uuid = self.thin_pool.create_filesystem(name, &dm, None)?;
-            result.push((*name, fs_uuid));
+        let mut names_iter = names.into_iter();
```

Comment:
Probably you want ```split_first``` here if you stick to this algorithm.

Reply:
`names` is a HashSet not a slice so alas it's not available.